python/lsst/source/injection/consolidate_injected_catalogs.py

```diff
@@ -0,0 +1,283 @@
+        for band in bands[1:]:
+            # make a column for the new band
+            multiband_catalog.add_column([np.nan] * len(multiband_catalog), name=f"{band}_{mag_suffix}")
+            # match the input catalog for this band to the existing
+            # multiband catalog
+            catalog_next_band = catalog_dict[band]
+            catalog_next_band.rename_column(self.config.col_mag, f"{band}_{mag_suffix}")
+            with Matcher(multiband_catalog[self.config.col_ra], multiband_catalog[self.config.col_dec]) as m:
+                idx, multiband_match_inds, next_band_match_inds, dists = m.query_radius(
+                    catalog_next_band[self.config.col_ra],
+                    catalog_next_band[self.config.col_dec],
+                    match_radius,
+                    return_indices=True,
+                )
+            # if there are matches...
+            if len(multiband_match_inds) > 0 and len(next_band_match_inds) > 0:
+                # choose the coordinates in the brightest band
+                for i, j in zip(multiband_match_inds, next_band_match_inds):
+                    mags = []
+                    for col in multiband_catalog.colnames:
+                        if f"_{mag_suffix}" in col:
+                            mags.append((col, multiband_catalog[i][col]))
+                    bright_mag = min([x[1] for x in mags])
+                    if catalog_next_band[f"{band}_{mag_suffix}"][j] < bright_mag:
+                        multiband_catalog[self.config.col_ra][i] = catalog_next_band[self.config.col_ra][j]
+                        multiband_catalog[self.config.col_dec][i] = catalog_next_band[self.config.col_dec][j]
+                # TODO: Once multicomponent object support is added, make some
+                # logic to pick the correct source_type.
+                # Fill the new mag value.
+                multiband_catalog[f"{band}_{mag_suffix}"][multiband_match_inds] = catalog_next_band[
+                    f"{band}_{mag_suffix}"
+                ][next_band_match_inds]
+                # add rows for all the sources without matches
+                not_next_band_match_inds = np.full(len(catalog_next_band), True, dtype=bool)
+                not_next_band_match_inds[next_band_match_inds] = False
+                multiband_catalog = vstack([multiband_catalog, catalog_next_band[not_next_band_match_inds]])
+            # otherwise just stack the tables
+            else:
+                multiband_catalog = vstack([multiband_catalog, catalog_next_band])
+        # fill any automatically masked values with NaNs
+        if multiband_catalog.has_masked_columns:
+            for col in multiband_catalog.columns:
+                if isinstance(multiband_catalog[col], MaskedColumn):
+                    multiband_catalog[col] = multiband_catalog[col].filled(np.nan)
+        return multiband_catalog
+
+    def run(
+        self,
+        catalog_dict: dict,
+        skymap: RingsSkyMap,
```

Comment:
Does this only work for a RingsSkyMap? Perhaps this should be BaseSkyMap, from lsst.skymap import BaseSkyMap.

Reply:
Thanks, fixed it!